pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -141,97 +137,103 @@ private void scanSegmentMetadataAndPurge() {
-        LOGGER.info("Table config null for table: {}, treating it as refresh only table.", offlineTableName);
-        return tableToDeletionStrategyMap;
+      offlineTableConfig =
+          ZKMetadataProvider.getOfflineTableConfig(_pinotHelixResourceManager.getPropertyStore(), offlineTableName);
+      if (offlineTableConfig == null) {
+        LOGGER.error("Table config is null, skip updating deletion strategy for table: {}.", offlineTableName);
+        return;
       }
-
-      SegmentsValidationAndRetentionConfig validationConfig = offlineTableConfig.getValidationConfig();
-
-      if (validationConfig.getSegmentPushType() == null || validationConfig.getSegmentPushType().isEmpty()) {
-        LOGGER.info("Segment push type for table {} is empty, skipping retention processing", offlineTableName);
-        return tableToDeletionStrategyMap;
-      }
-
-      if ("REFRESH".equalsIgnoreCase(validationConfig.getSegmentPushType())) {
-        LOGGER.info("Table: {} is a refresh only table.", offlineTableName);
-        return tableToDeletionStrategyMap;
-      }
-
-      LOGGER.info("Building retention strategy for table {} with configuration {}", offlineTableName, validationConfig);
-
-      final String retentionTimeUnit = validationConfig.getRetentionTimeUnit();
-      final String retentionTimeValue = validationConfig.getRetentionTimeValue();
-
-      if (retentionTimeUnit == null || retentionTimeUnit.isEmpty() || retentionTimeValue == null || retentionTimeValue.isEmpty()) {
-        LOGGER.info("Table {} has an invalid retention period of {} {}, skipping", offlineTableName,
-            retentionTimeValue, retentionTimeUnit);
-        return tableToDeletionStrategyMap;
-      }
-
-      TimeRetentionStrategy timeRetentionStrategy = new TimeRetentionStrategy(retentionTimeUnit,
-          retentionTimeValue);
-      tableToDeletionStrategyMap.put(offlineTableName, timeRetentionStrategy);
     } catch (Exception e) {
-      LOGGER.error("Error creating TimeRetentionStrategy for table: {}", offlineTableName, e);
+      LOGGER.error(
+          "Caught exception while getting table config from property store, skip updating deletion strategy for table: {}.",
+          offlineTableName, e);
+      return;
     }
 
-    return tableToDeletionStrategyMap;
-  }
+    // Fetch validation config.
+    SegmentsValidationAndRetentionConfig validationConfig = offlineTableConfig.getValidationConfig();
+    if (validationConfig == null) {
+      LOGGER.error("Validation config is null, skip updating deletion strategy for table: {}.", offlineTableName);
+      return;
+    }
 
-  private Map<String, RetentionStrategy> handleRealtimeDeletionStrategy(String realtimeTableName) {
-    Map<String, RetentionStrategy> tableToDeletionStrategyMap = new HashMap<String, RetentionStrategy>();
+    // Fetch push type.
+    if ((validationConfig.getSegmentPushType() == null) || (!validationConfig.getSegmentPushType()
```

Comment:
What if there's a typo in the segmentPushType (ie neither APPEND or REFRESH)? We should throw an exception in that case, right?